### PR TITLE
Add CI/CD actions

### LINKED ISSUE #11: Create CI -  CD
This issue is part of Epic #2 

## Objective

Create a CI CD for the project.

## Checklist

- [ ] Create a CI for the project
- [ ] Create a CD for the project

### PR DESCRIPTION
**Pull request by Odraxs**
## Description

Some files required updates due to warnings generated by the `cargo clippy —all-targets —all-features — -D warnings` command.

Fixes #11 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings

### CODE REVIEW

File: .github/workflows/cd.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: Smart Contract Deployment
+
+on:
+  pull_request:
+    types: [closed]
+    branches:
+      - main
+      - staging
```
Comment (einerzg) [inline .github/workflows/cd.yaml:8]: here it should be something like `"v*.[0-9]"`. to handle versions

File: .github/workflows/cd.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          --network-passphrase 'Test SDF Future Network ; October 2022'
+      - name: Contract Deployment to mainnet
+        if: ${{ github.event.pull_request.base.ref }} == main
+        ## Change when the mainnet for soroban gets released
+        run: |
+          soroban deploy \
+          --wasm target/wasm32-unknown-unknown/release/soroban_devops.wasm \
+          --secret-key ${{ secrets.DEPLOYER_SECRET }} \
+          --rpc-url https://horizon-futurenet.stellar.cash:443/soroban/rpc \
+          --network-passphrase 'Test SDF Future Network ; October 2022'
```
Comment (einerzg) [inline .github/workflows/cd.yaml:50]: I understand that so far only the `futurenet` is available.

File: mtk-automation/src/contract_actions/member.rs [hunk 3]
```diff
@@ -11,14 +11,14 @@ use soroban_auth::Signature;
 use soroban_sdk::{vec, AccountId, Env, RawVal, Vec};
```
Comment (einerzg) [inline mtk-automation/src/contract_actions/member.rs:11]: @Odraxs, to fix this file should be a different PR.

**Review by einerzg — CHANGES_REQUESTED** [review]

File: mtk-automation/src/contract_actions/member.rs (hunk 3 above)
Comment (Odraxs) [reply mtk-automation/src/contract_actions/member.rs:11]: @einerzg Ok
But IDK if GitHub allows merging the code that fails those checks, is it possible?

Comment (einerzg) [reply mtk-automation/src/contract_actions/member.rs:11]: Well @Odraxs, forget this comment then. Next time add in the PR the info on why this file is there.

File: .github/workflows/cd.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+name: Smart Contract Deployment
+
+on:
+  pull_request:
+    types: [closed]
+    branches:
+      - main
+      - staging
```
Comment (einerzg) [inline .github/workflows/cd.yaml:8]: For this project is not necessary the `staging` branch. Please remove

File: .github/workflows/cd.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+          --network-passphrase 'Test SDF Future Network ; October 2022'
+      - name: Contract Deployment to mainnet
+        if: ${{ github.event.pull_request.base.ref }} == main
+        ## Change when the mainnet for soroban gets released
+        run: |
+          soroban deploy \
+          --wasm target/wasm32-unknown-unknown/release/soroban_devops.wasm \
+          --secret-key ${{ secrets.DEPLOYER_SECRET }} \
+          --rpc-url https://horizon-futurenet.stellar.cash:443/soroban/rpc \
+          --network-passphrase 'Test SDF Future Network ; October 2022'
```
Comment (einerzg) [inline .github/workflows/cd.yaml:51]: At the moment we only handle the deploy for the futurenet. Please remove this.

**Review by einerzg — CHANGES_REQUESTED** [review]

**Review by einerzg — APPROVED** [review]
